core/src/test/java/org/elasticsearch/index/analysis/AnalysisModuleTests.java

```diff
@@ -275,4 +281,25 @@ public void testUnderscoreInAnalyzerNameAlias() {
             assertThat(e.getCause().getMessage(), equalTo("analyzer name must not start with '_'. got \"_invalid_name\""));
         }
     }
+
+    @Test
+    public void testBackwardCompatible() {
+        String yaml = "/org/elasticsearch/index/analysis/test1.yml";
```

Comment:
I don't see why this needs to be a separate file. Just construct json here in the test?